### PR TITLE
#164694217 Users should receive an email notification on successful signup

### PR DESCRIPTION
**Pull request by sebbss**
#### What does this PR do?
- email verification link.
#### Description of Task to be completed?
- Implement sending email verification link after successful signup 
-Implement Activate user account after clicking the verification link in the email
#### How this should be manually tested
* After running application server with the command `python manage.py runserver`
- startup postman service to consume `http://127.0.0.1:8000/api/users/` endpoint
- using the `POST` method send JSON data in the body
```
{"user":{
"username":"value",
"email":"value",
"password":"value"}
```
*After successful signup check your email for verification link
*click the verification link and you will be verified
#### Screen shots
image

#### What are the relevant pivotal tracker stories?
[#164694217](https://www.pivotaltracker.com/story/show/164694217)

### CODE REVIEW

**Comment by kimbugp** [comment]
@sebbss , could you please use the wiki to make commits.

**Comment by kimbugp** [comment]
@sebbss, ensure that the PTboard id in the PR description is linked to the story URL

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -28,9 +31,31 @@ def post(self, request):
...
-
+        email = serializer.data['email']
+        verification_token = RegistrationAPIView.verification_token(email)
+        RegistrationAPIView.verification_link(email,verification_token)
         return Response(serializer.data, status=status.HTTP_201_CREATED)
 
+    @staticmethod
+    def verification_link(email,token):
+        subject = "Activation for your account"
+        message = f'Please Activate your account below.\n{token}'
```
Comment (kimbugp) [inline authors/apps/authentication/views.py]: @sebbss, the body of the email should reference a URL which can be clicked to activate the account.

**Comment by kimbugp** [comment]
@sebbss , could you add a description of how your PR can be tested

**Comment by kimbugp** [comment]
@sebbss , how are you limiting a user from logging in when they are not verified.

**Review by kimbugp — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (kimbugp) [inline .vscode/settings.json]: @sebbss , this file should not be pushed to github

**Review by kimbugp — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/tests/test_views.py [hunk 3]
```diff
@@ -40,3 +40,25 @@ def test_login_with_invalid_user_fails(self):
...
+    def test_login_user_login_token(self):
+        user=User.objects.create_user(username='sampleuser', email='user@sprinters.ug',password='butt3rfly1')
+        user.is_active=True
+        user.save()
+
+        response = self.client.post(
+            reverse('auth:login'),
+            data=self.user_data,
+            format='json')
+        self.assertIn('token', response.data)
```
Comment (Rhytah) [inline authors/apps/authentication/tests/test_views.py]: @sebbss the variable `token` is not defined anywhere so this test inevitably fails. Consider changing to `auth_token` that is defined in `serializers.py` and `models.py` for token.
Same goes for `generate_auth_token`

**Comment by kimbugp** [comment]
@sebbss , ensure your commit uses the team conventions

File: authors/apps/authentication/tests/test_views.py (hunk 3 above)
Comment (sebbss) [reply authors/apps/authentication/tests/test_views.py]: i have rebased with develop and made all the tests pass

File: .vscode/settings.json (hunk 2 above)
Comment (sebbss) [reply .vscode/settings.json]: i removed it, pushed it by mistake

**Comment by kimbugp** [comment]
@sebss , could you ensure that you don't lower the test coverage

**Review by kimbugp — APPROVED** [review]

**Pull request merged**